### ISSUE TITLE
Don't accept inbound connection establishments more than `max_peering_degree`

### ISSUE THREAD

**Issue by youngjoon-lee** [logos-co/nomos-node#939]
If a node already has the `max_peering_degree` number of connections, the node should reject additional inbound connection establishments to save bandwidth. 
This has been discussed in the [comment](https://www.notion.so/Nomos-Blend-Network-Tier-1-Persistent-Transmission-Module-10b8f96fb65c807cb1e8f92a7f41a771?d=1518f96fb65c8009b2a8001c9b91945e&pvs=4#14b8f96fb65c80028981dd3c6027a3a4) and added in the [specification](https://www.notion.so/Nomos-Blend-Network-Tier-1-Persistent-Transmission-Module-10b8f96fb65c807cb1e8f92a7f41a771?pvs=4#1588f96fb65c8014bc9cc6cc14431c3c) at 2024-12-10.

**Comment by danielSanchezQ** [comment]
> If a node already has the `peering_degree` number of healthy connections, the node should reject additional inbound connection establishments to save bandwidth. This hasn't been specified in the specification, but recently discussed in the [comment](https://www.notion.so/Nomos-Blend-Network-Tier-1-Persistent-Transmission-Module-10b8f96fb65c807cb1e8f92a7f41a771?d=1518f96fb65c8009b2a8001c9b91945e&pvs=4#14b8f96fb65c80028981dd3c6027a3a4). Here, we shouldn't use the `max_peering_degree` parameter, which is for limiting additional "outbound" connection establishments to replace malicious/unhealthy connections, not for limiting inbound connection establishments.


I thought connections were used for both sending and receiving meaning that blendnodes would have a total of `max_peering_degree` connections. This kinda makes sense, if a peer is missbehaving then we need to remove **both** incoming and outgoing (streams, as the connection should be the same?).

**Comment by youngjoon-lee** [comment]
> I thought connections were used for both sending and receiving meaning that blendnodes would have a total of `max_peering_degree` connections. This kinda makes sense, if a peer is missbehaving then we need to remove **both** incoming and outgoing (streams, as the connection should be the same?).

Exactly. I developed the discussion with Marcin and update the specification. Accordingly, I updated the issue description above. Yes. All connections are bidirectional. Each node must have `max_peering_degree` connections at most.
